sdk-cpp/include/do_exceptions.h

```diff
@@ -1,21 +1,46 @@
 #ifndef _DELIVERY_OPTIMIZATION_DO_EXCEPTIONS_H
 #define _DELIVERY_OPTIMIZATION_DO_EXCEPTIONS_H
 
+#if (!DO_DISABLE_EXCEPTIONS)
+
 #include <exception>
 #include <stdint.h>
 #include <system_error>
 
-#if defined(DO_INTERFACE_COM)
-#include <winerror.h>   // FAILED macro
-#endif
+#endif // !DO_DISABLE_EXCEPTIONS
 
 namespace microsoft
 {
 namespace deliveryoptimization
 {
 
+// Error Macros
+#ifndef RETURN_HR_IF_FAILED
+#define RETURN_HR_IF_FAILED(hr)  {  \
+    int32_t __hr = (hr);            \
+    if(FAILED(__hr)) return __hr;   \
+}
+#endif
+
+#ifndef FACILITY_DELIVERY_OPTIMIZATION 
+#define FACILITY_DELIVERY_OPTIMIZATION   208
+#endif
+
+#define HRESULT_FROM_SYSTEM_ERROR(x) ((int32_t)(x) <= 0 ? ((int32_t)(x)) : ((int32_t) (((x) & 0x0000FFFF) | (FACILITY_DELIVERY_OPTIMIZATION << 16) | 0x80000000)))
+
```

Comment:
Actually, now I'm not so sure that this is a good idea. It'll kind of morph the error code and become harder to understand what it corresponds to.

I think what we can do is define all error codes in the errc enum below.